Source/drlg_l3.cpp

```diff
@@ -1180,8 +1182,8 @@ void DRLG_L3PoolFix()
 {
 	int dunx, duny;
 
-	for (duny = 0; duny < DMAXY; duny++) {
-		for (dunx = 0; dunx < DMAXX; dunx++) {
+	for (duny = 0; duny < DMAXY; duny++) { // BUGFIX: Change 'DMAXY' to 'DMAXY - 1'

```

Comment:
I think the fix is to start at `duny = 1` and `dunx = 1`, rather than using `DMAXY - 1`. Otherwise, `dungeon[dunx - 1][duny - 1]` would be out of bounds.